### PR TITLE
Handle null body in Task and Comment model

### PR DESCRIPTION
**Pull request by fxvarga**
# What's in this PR?
Handle null body in Task and Comment model by showing the message "No  description provided" in the comment item and task item templates.
## References
Fixes [#1385](https://github.com/code-corps/code-corps-ember/issues/1385)

### CODE REVIEW

File: tests/pages/components/task-details.js [hunk 1]
```diff
@@ -22,7 +22,10 @@ export default {
     hasStrongText: isVisible('strong'),
     text: text()
   },
-
+  nullCommentBody:{
+    scope: '.null-comment-body',
```
Comment (JoshSmith) [inline tests/pages/components/task-details.js]: I would rename this class to `data-test-null-body` (in all cases here) which will get automatically removed later. Also think slightly less confusing than `comment` in context of task.

File: tests/integration/components/task-details-test.js [hunk 2]
```diff
@@ -133,3 +133,15 @@ test('it saves', function(assert) {
   page.save.click();
   assert.equal(page.commentBody.text, 'A new body', 'The body is saved');
 });
+test('if the task is null render the no description element', function(assert) {
```
Comment (JoshSmith) [inline tests/integration/components/task-details-test.js]: Would add a space above this test.

File: tests/integration/components/task-details-test.js [hunk 3]
```diff
@@ -133,3 +133,15 @@ test('it saves', function(assert) {
...
+  stubService(this, 'mention-fetcher', {
+    prefetchBodyWithMentions() {
+      return null;
+    }
+  });
+
+  this.set('task', mockTask);
+
+  page.render(hbs`{{task-details task=task}}`);
+  assert.ok(page.nullCommentBody.isVisible, 'The message for no comment body is rendered');
```
Comment (JoshSmith) [inline tests/integration/components/task-details-test.js]: Can we test for the opposite case where there _is_ a body (i.e. this should _not_ render)?

File: app/components/comment-item.js [hunk 4]
```diff
@@ -62,13 +62,30 @@ export default Component.extend({
...
+   * 
+   * checks to see if outcome of processed content is null, indicating null comment body
+   * and returns computed property as true if body is null.
+   * 
+   * NOTE: only did it this way because there is a test that lead me to think that this
+   * component could possibly load before comment body is loaded. This will prevent the 
+   * "No description provided" message from showing up on load before data is available.
+   */
+  nullBody: computed('processedBody', function() {
+    if (get(this, 'processedBody') === null) {
```
Comment (JoshSmith) [inline app/components/comment-item.js]: I think we may want to use `Ember.isBlank` here.

File: app/components/comment-item.js [hunk 5]
```diff
@@ -62,13 +62,30 @@ export default Component.extend({
    * @property processedComment
    @ @type String
    */
-  processedBody: computed('comment.body', 'mentions', function() {
+  processedBody: computed('comment.body', 'comment.isLoaded', 'mentions', function() {
     let body = get(this, 'comment.body');
     let mentions = get(this, 'mentions');
-    return parse(body, mentions || []);
+    let parsedBody = parse(body, mentions || []);
+    return parsedBody;
```
Comment (JoshSmith) [inline app/components/comment-item.js]: Also wanted to say I think we should just still write this as `return parse`. Is there a reason you rewrote this?

Comment (JoshSmith) [reply app/components/comment-item.js]: Maybe we should rewrite `processedBody` to `parsedBody` generally? Any thoughts @begedin since I think you wrote this originally?

File: tests/pages/components/task-details.js (hunk 1 above)
Comment (JoshSmith) [reply tests/pages/components/task-details.js]: Whoops. So this is not a CSS class I'm looking for but a `data-` attr. Just to clarify.

File: app/components/comment-item.js (hunk 5 above)
Comment (begedin) [reply app/components/comment-item.js]: Both seem to fit fine so I don't lean towards any of the names.

Comment (JoshSmith) [reply app/components/comment-item.js]: I think that case then if we can change `processedBody` to `parsedBody` that'd be easier to grok.

File: tests/pages/components/task-details.js (hunk 1 above)
Comment (begedin) [reply tests/pages/components/task-details.js]: @JoshSmith You sure you don't want a class to, for potential styling?

Github, for example, seems to italicize the text and make it a bit lighter than the normal body text.

Comment (fxvarga) [reply tests/pages/components/task-details.js]: To verify, is this what you are looking for?

```<em class="null-comment-body" data-test-null-body=true>No description provided.</em>```

Comment (JoshSmith) [reply tests/pages/components/task-details.js]: `<em data-test-null-body>No description provided.</em>`

Comment (JoshSmith) [reply tests/pages/components/task-details.js]: In `/styles/_markdown.scss` I would add

```css
.markdown-body--empty {
  em {
    color: $text--light;
  }
}
```

**Comment by fxvarga** [comment]
Added another commit to this pull request, with the changes for above.

File: tests/integration/components/task-details-test.js [hunk 6]
```diff
@@ -145,3 +146,16 @@ test('if the task is null render the no description element', function(assert) {
   page.render(hbs`{{task-details task=task}}`);
   assert.ok(page.nullCommentBody.isVisible, 'The message for no comment body is rendered');
 });
+
+test('if the task body is not nul do not render the no description element', function(assert) {
```
Comment (JoshSmith) [inline tests/integration/components/task-details-test.js]: `not nul` to `not null` here.

File: app/components/comment-item.js [hunk 7]
```diff
@@ -72,15 +75,16 @@ export default Component.extend({
...
+   * checks to see if outcome of processed content is null, indicating null 
+   * comment body and returns computed property as true if body is null.
    * 
-   * NOTE: only did it this way because there is a test that lead me to think that this
-   * component could possibly load before comment body is loaded. This will prevent the 
-   * "No description provided" message from showing up on load before data is available.
+   * It tests specifically against null because parsedBody generates other 
+   * falsey outputs even when comment body is not null. 
+   * For example is if comment data is not loaded yet, 
+   * parsedBody trurns undefined
```
Comment (JoshSmith) [inline app/components/comment-item.js]: Would reword as:

> It tests specifically against `null` because `parsedBody` generates other falsey outputs even when comment body is not `null`. For example, if the comment data is not loaded yet, `parsedBody` returns `undefined`.

File: tests/pages/components/task-details.js [hunk 8]
```diff
@@ -22,7 +22,10 @@ export default {
     hasStrongText: isVisible('strong'),
     text: text()
   },
-
+  nullCommentBody:{
+    scope: '.markdown-body--empty',
```
Comment (JoshSmith) [inline tests/pages/components/task-details.js]: You'll want to change this and the other `scope` to `[data-test-markdown-body-empty]`, and you can remove the `isVisible` and call `isVisible()` directly in the test without issue.

File: app/styles/_markdown.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+.markdown-body--empty {
+    color: $text--light;
+}
```
Comment (JoshSmith) [inline app/styles/_markdown.scss]: You can remove this file and rebase on `upstream/develop` which will pull this file in and add the class there. I forgot you were not rebased yet onto our `upstream/develop`.

File: app/styles/app.scss [hunk 10]
```diff
@@ -54,7 +54,7 @@
 @import "code";
 @import "images";
 @import "inputs";
-
+@import "markdown";
```
Comment (JoshSmith) [inline app/styles/app.scss]: This line can be reverted.

**Comment by JoshSmith** [comment]
Comment.
